NuCivic/dkan
# Update harvest source messages CIVIC-6644 (1.13.7 version)

ssue: CIVIC-6644

## Description
When using the filter or exclude function when creating a harvest source, an error message will display for any dataset that does not have a value for the key being filtered/excluded. For example if you filter by 'theme' any dataset that does not have a value for 'theme' will display this error:
 **Error trying to access to key @theme. Please make sure @theme exists in the dataset**
and if there are 50 datasets without a value for theme, it will display 50 error messages.

## Note
If you filter by a field where none of the datasets have a value, you will get all datasets listed in the preview.

## Improvements
**Filter**
Current
`Error trying to access to key @path. Please make sure @path exists in the dataset`
New
`Datasets that do not have a value for @path will not be included in the harvest.`

**Exclude**
Current
`Error trying to access to key @path. Please make sure @path exists in the dataset`
New
`Datasets that do not have a value for @path will be included in the harvest.`

And only a single warning message is displayed, not an error for each instance.

**Override**
Current
`Key "@path" was automatically generated because doesn\'t exists in the dataset "@title". Make sure you want to modify the key "@path".`
New Default
`The @path field was set to "@value" on all harvested datasets.`

**Default**
Current
`Key "@path" was automatically generated because doesn\'t exists in the dataset "@title". Make sure you want to modify the key "@path".`
New Default
`A @path value of "@value" was added to datasets where the @path field was empty.`

## QA Tests
1. Create a harvest source from a source you know does not have values for theme on all datasets.
2. Add a filter (accrualPeriodicity = nada)
3. Add a default (theme = health)
4. Add an override (granularity = weekly)
5. Confirm you only see one warning message about accrualPeriodicity.
6. The preview should be empty
7. Now change accrualPeriodicity = R/P1Y
8. Confirm you see messages about the default and override

## To Do
Create a new ticket to refactor this process, the function is too long.


## Reminders
